core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataDeleteIndexService.java

```diff
@@ -168,28 +168,27 @@ public void onResponse(final Response response) {
         public void onFailure(Throwable t) {
             if (notified.compareAndSet(false, true)) {
                 FutureUtils.cancel(future);
-                listener.onFailure(t);
+                resultListener.onFailure(t);
             }
         }
     }
 
+    public interface Listener<T> {
```

Comment:
nit, and also my personal opinion: I think the generics type here makes the code less readable, given that at the end  you do need the Response in the transport action. I don't think this change is worthwhile, as it only delays the creation of the Response object. Makes sense?